### PR TITLE
add json array sorting tests

### PR DESCRIPTION
**Pull request by Jinxiao0302**
Related to #1411 

This PR creates a testing table with a column of json array type, and test cases for json array sorting behavior.

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `master` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by mathemancer — CHANGES_REQUESTED** [review]
First, please use our custom JSON array type for the table so it's clear what the goals of the test are. Also, please add some nested structures to the test data so we can show how they sort as well. Finally, I think you should specify the entire expected array so that whoever is reading your test in the future can reason through the rule set for sorting this type.

**Comment by Jinxiao0302** [comment]
> First, please use our custom JSON array type for the table so it's clear what the goals of the test are. Also, please add some nested structures to the test data so we can show how they sort as well. Finally, I think you should specify the entire expected array so that whoever is reading your test in the future can reason through the rule set for sorting this type.

Thanks for your reviews. I have made changes to the code following the instructions. 
For the nested arrays, the nested arrays come later than other types of elements when compared. When comparing among nested arrays,  the same rule of outer array ordering is used.

File: db/tests/records/operations/test_sort.py [hunk 1]
```diff
@@ -21,6 +21,16 @@ def test_get_records_gets_ordered_records_num_col(roster_table_obj):
...
 
 
+def test_json_sort_array(json_table_obj):
+    roster, engine = json_table_obj
+    order_list = [{"field": "json_array", "direction": "asc"}]
+    record_list = get_records(roster, engine, order_by=order_list)
+    assert record_list[0][0] == 1 and record_list[1][0] == 4 and record_list[2][0] == 12 and record_list[3][0] == 14 \
+        and record_list[4][0] == 13 and record_list[5][0] == 2 and record_list[6][0] == 6 and record_list[7][0] == 8 \
+        and record_list[8][0] == 9 and record_list[9][0] == 7 and record_list[10][0] == 10 and record_list[11][0] == 11 \
+        and record_list[12][0] == 5 and record_list[13][0] == 3
```
Comment (mathemancer) [inline db/tests/records/operations/test_sort.py:31]: The goals of this test should be both checking the behavior, but also providing an example of how sorting will look.

```suggestion
    assert [row["json_array"] for row in record_list] == [
        '[]',
        '["BMW", "Ford", "Fiat"]',
        '["BMW", "Ford", [1, 2]]',
        '["BMW", "Ford", ["Akshay", "Prashant", "Varun"]]',
        '["BMW", "Ford", [1, 2, 3]]',
        '["Ford", "BMW", "Fiat"]',
        '[1, 2, 3]',
        '[1, 2, false]',
        '[1, 2, true]',
        '[2, 3, 4]',
        '[false, false, false]',
        '[true, true, false]',
        '["BMW", "Ford", "Fiat", "Fiat"]',
        '["Ram", "Shyam", "Radhika", "Akshay", "Prashant", "Varun"]',
    ]
```

Please double-check my ordering and types for the suggestion. I hope it makes clear how the test should look, though.

**Review by mathemancer — CHANGES_REQUESTED** [review]
See my specific suggestion. Good work so far!

**Review by mathemancer — APPROVED** [review]
This looks great. Thanks @Jinxiao0302 !

**Pull request merged**